frame/dex/src/lib.rs

```diff
@@ -433,19 +417,73 @@ pub mod pallet {
 			})
 		}
 
+		/// some native --> exact number of tokens
 		#[pallet::weight(0)]
-		pub fn swap_exact_tokens_for_tokens(
+		pub fn buy_exact_tokens_for_native(
```

Comment:
```suggestion
		pub fn buy_exact_tokens(
```
no need to put `for_native` since there are no other options